app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseViewModel.kt

```diff
@@ -116,6 +121,38 @@ class MyCourseViewModel(
         }
     }
 
+    fun fetchExtensions(productId: String?) {
+        runIO {
+            when (val response = productId?.let { repo.getExtensions(Integer.parseInt(it)) }) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        body()?.let {
+                            extensions.postValue(it.productExtensions)
+                        }
+                    } else {
+                        setError(fetchError(code()))
+                    }
+                }
+            }
+        }
+    }
+
+    fun buyExtension(extensionId: Int?) {
+        runIO {
+            when (val response = extensionId?.let { repo.buyExtension(it) }) {
+                is ResultWrapper.GenericError -> setError(response.error)
+                is ResultWrapper.Success -> with(response.value) {
+                    if (isSuccessful) {
+                        purchase.postValue(true)
```

Comment:
@aggarwalpulkit596 bhaiya, for buying functionality, we can observe `purchase` in the Overview fragment and do proceedings. But what should be the flow from there onwards? Please let me know bhaiya.